File: csharp/Database/ApplicationDatabase/Models/Account.cs
namespace Database.ApplicationDatabase.Models;

public class Account
{
    public Guid Id;
    public string? ExternalId;
    public string? Source;
    public required string Name;
    public required string Email;
    public required Role Role;
    public DateTime CreatedAt = DateTime.UtcNow;
    public DateTime UpdatedAt = DateTime.UtcNow;
}

File: backend/lib/auth_server/dal/models/Account.go
package models

import (
	"backend/pkg/core/models"
	"time"

	"github.com/google/uuid"
)

type Account struct {
	Id         uuid.UUID
	ExternalId string
	Source     string
	Name       string
	Email      string
	Role       models.Role
	CreatedAt  time.Time
	UpdatedAt  time.Time
}
